core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/RateAppCounter.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.utils.dialog
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class RateAppCounter internal constructor(context: Context) {
+  private var visitCounter: SharedPreferences
+
+  init {
+    visitCounter = context.getSharedPreferences(MASTER_NAME, 0)
+    visitCounter = context.getSharedPreferences(NO_THANKS_CLICKED, 0)
+  }
+
+  var noThanksState: Boolean
+    get() = visitCounter.getBoolean(NO_THANKS_CLICKED, false)
+    set(value) {
+      visitCounter.edit().apply {
+        putBoolean(NO_THANKS_CLICKED, value)
+        apply()
```

Comment:
As we are using ktx extensions, ti'd be better to see this written via advantage of their brevity and cleanness